orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/helpers/ConstructUriHelper.java

```diff
@@ -1,40 +1,64 @@
 package uk.gov.di.orchestration.shared.helpers;
 
-import org.apache.http.client.utils.URIBuilder;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hc.core5.net.URIBuilder;
 
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.util.Map;
-import java.util.Objects;
+import java.util.Optional;
 
 public class ConstructUriHelper {
 
+    public static URI buildURI(URI baseUrl, String path, Map<String, String> queryParams) {
+        return buildURI(baseUrl.toString(), path, queryParams);
+    }
+
     public static URI buildURI(String baseUrl, String path, Map<String, String> queryParams) {
-        if (!baseUrl.endsWith("/")) {
-            baseUrl = baseUrl + "/";
-        }
-        var uri =
-                Objects.isNull(path)
-                        ? URI.create(baseUrl)
-                        : URI.create(baseUrl + path.replaceAll("^/+", ""));
+        return buildURI(baseUrl, Optional.ofNullable(path), Optional.ofNullable(queryParams));
+    }
+
+    public static URI buildURI(URI baseUri, Map<String, String> queryParams) {
+        return buildURI(baseUri.toString(), queryParams);
+    }
+
+    public static URI buildURI(String baseUri, Map<String, String> queryParams) {
+        return buildURI(baseUri, Optional.empty(), Optional.ofNullable(queryParams));
+    }
+
+    public static URI buildURI(URI baseUri, String path) {
+        return buildURI(baseUri.toString(), path);
+    }
+
+    public static URI buildURI(String baseUri, String path) {
+        return buildURI(baseUri, Optional.ofNullable(path), Optional.empty());
+    }
+
+    public static URI buildURI(URI baseUri) {
+        return buildURI(baseUri.toString());
+    }
```

Comment:
How many of these methods are being used?

Reply:
I believe almost all of them. I think build URI with a base URI only may only be used in the test for this class, but that was already there.

Reply:
I've removed the overloads with the base URI only.